force_wfmanager/wfmanager_setup_task.py

```diff
@@ -453,10 +452,13 @@ def _server_event_mainthread(self, event):
         action appropriately according to the type"""
         if isinstance(event, MCOStartEvent):
             self.analysis_model.clear()
-            self.analysis_model.value_names = tuple(event.serialize())
             self.computation_running = True
-        elif isinstance(event, MCOProgressEvent):
-            self.analysis_model.add_evaluation_step(event.serialize())
+        try:
+            event_data = event.serialize()
+        except AttributeError:
+            return
+
+        self.analysis_model.notify(event_data)
```

Comment:
At the moment, could the `analysis_model.notify` method handle any data that didn't come from a `MCOProgressEvent`? If not, then we should probably keep this dependency hardcoded here, and aim to replace it with a more generic events handler in the future.

Reply:
The `notify` method is a way to insert data into the current row of the `AnalysisModel`. The `analysis_model.notify` can handle two formats:
- `{column_label: data_to_add}`
- `(list / tuple of data_to_add)`
The `AnalysisModel` knows nothing about events and their types.

Reply:
Followed by the offline discussion, it is clear that implementing an event handler is very important here.
The main point is that the `AnalysisModel` should somehow perform "data filtering" before storing the data.
A possible solution would be to 
- Move the implementation of the `AnalysisModel` to the BDSS;
- Continue to instantiate the `AnalysisModel` in the workflow manager;
- Implement `MCOEvent`s preprocessing as a part of the `AnalysisModel.notify`.

Reply:
This has been resolved for the time being by keeping the hard coded filtering of event types in the `WfManagerSetupTask` so that the data handling is secure, whilst commenting that we should move the logic onto the `AnalysisModel` in future.